sdk/core/System.ClientModel/src/Options/ClientPipelineOptions.cs

```diff
@@ -161,7 +200,11 @@ internal static PipelinePolicy[] AddPolicy(PipelinePolicy policy, PipelinePolicy
     /// instance or call methods that would change its state will throw
     /// <see cref="InvalidOperationException"/>.
     /// </summary>
-    public virtual void Freeze() => _frozen = true;
+    public virtual void Freeze()
```

Comment:
outside the scope of this PR, but I think it's wild that we depend on immutability but I could extend the type and refuse to call the base class, thus rendering `Freeze` unable to, you know, freeze things.

Reply:
Yeah that's a good point

Reply:
Maybe it's a "feature"... Could customers technically use this to allow in-flight options changes... ?